Add Prettier extension to remote container

ajyong opened the pull request:
`devcontainer.json` got formatted by... prettier.

ajyong committed:
Add Prettier extension to remote container

- devcontainer.json got formatted by... prettier.

MarkBennett reviewed (changes requested):
Cool idea @ajyong 

Can you just clarify when this will format to code? With this installed are files formatted on save, or do you need to trigger the add-on?

ajyong commented:
By default it does nothing.  I think I should add some instructions in the README to:

1. Make Prettier the default formatter
1. Clarify that it formats per-file, on-save if this is enabled in VS Code

ajyong committed:
Add style guide settings to remote container

- Trim whitespace (except Markdown)
- Format on save using Prettier
- Use Prettier as default formatter
- Add documentation to README

ajyong commented:
It's more heavy-handed now, and I've added documentation in the README to reflect this.  By default, this PR sets the remote container to:

- Use Prettier as the **default formatter**
- Format your files **on save**
- Trim trailing whitespace from all files **except Markdown**

Give your container a rebuild and let me know if it impedes your workflow @MarkBennett, otherwise I think these "sane defaults" will increase consistency moving forward.

MandyMeindersma commented:
lgtm, I will let mark decide on final approval

MarkBennett commented on README.md:
```diff
@@ -20,11 +20,34 @@ This will serve the site at:
+
+By default, we have set the remote container to:
+
+- Use Prettier as the **default formatter**
```
Sounds good to me!

MarkBennett reviewed (approved):
These changes look like a good start with Prettier @ajyong. Let's try them out and revise if needed.
